frontend/src/pages/data-product/components/data-product-tabs/data-output-tab/components/dataset-table/dataset-table.component.tsx

```diff
@@ -0,0 +1,92 @@
+import { Button, Flex, Form, Typography } from 'antd';
+import { useMemo } from 'react';
+import { useTranslation } from 'react-i18next';
+import { Link } from 'react-router';
+import { DatasetCard } from '@/components/datasets/dataset-card/dataset-card.component';
+import { Searchbar } from '@/components/form';
+import { useCheckAccessQuery } from '@/store/features/authorization/authorization-api-slice.ts';
+import { useGetDataProductByIdQuery } from '@/store/features/data-products/data-products-api-slice.ts';
+import { AuthorizationAction } from '@/types/authorization/rbac-actions.ts';
+import type { DatasetsGetContract } from '@/types/dataset/datasets-get.contract.ts';
+import { ApplicationPaths } from '@/types/navigation.ts';
+import type { SearchForm } from '@/types/shared';
+import styles from './dataset-table.module.scss';
+
+type Props = {
+    dataProductId: string;
+    datasets: DatasetsGetContract;
+    isDragActive?: boolean;
+    draggedDataOutputId?: string | null;
+};
+
+export function DatasetTable({ dataProductId, datasets, isDragActive, draggedDataOutputId }: Props) {
+    const { t } = useTranslation();
+    const { data: dataProduct, isLoading: isLoadingDataProduct } = useGetDataProductByIdQuery(dataProductId);
+
+    const { data: access } = useCheckAccessQuery(
```

Comment:
Can we abstract this away in the backend? Has access to create dataset? 

At a minimum rename the variables to something more explicit.